### PR TITLE
[SPARK-43134] [CONNECT] [SS] JVM client StreamingQuery exception() API

### PR DESCRIPTION
**Pull request by WweiL**
### What changes were proposed in this pull request?

Add StreamingQuery exception() API for JVM client

### Why are the changes needed?

Development of SS Connect

### Does this PR introduce _any_ user-facing change?

Yes

### How was this patch tested?

Manual test:
```
Spark session available as 'spark'.
   _____                  __      ______                            __
  / ___/____  ____ ______/ /__   / ____/___  ____  ____  ___  _____/ /_
  \__ \/ __ \/ __ `/ ___/ //_/  / /   / __ \/ __ \/ __ \/ _ \/ ___/ __/
 ___/ / /_/ / /_/ / /  / ,<    / /___/ /_/ / / / / / / /  __/ /__/ /_
/____/ .___/\__,_/_/  /_/|_|   \____/\____/_/ /_/_/ /_/\___/\___/\__/
    /_/

@  val q = spark.readStream.format("rate").load().writeStream.option("checkpointLocation", "/home/wei.liu/ckpt").toTable("my_table")  
q: org.apache.spark.sql.streaming.StreamingQuery = org.apache.spark.sql.streaming.RemoteStreamingQuery@772f3a3f

@ q.exception 
res1: Option[org.apache.spark.sql.streaming.StreamingQueryException] = None

@ q.stop() 

```

### CODE REVIEW

**Comment by WweiL** [comment]
@rangadi @pengzhon-db

**Comment by WweiL** [comment]
Stack trace will be added in this ticket https://issues.apache.org/jira/browse/SPARK-43206, after this PR is merged

File: connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/streaming/StreamingQuery.scala
Comment (bogao007) [inline connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/streaming/StreamingQuery.scala:218]: Just curious, is the existing error framework accessible in Spark Connect jvm client?

**Review by bogao007 — APPROVED** [review]
LGTM

File: connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/streaming/StreamingQuery.scala
Comment (WweiL) [reply connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/streaming/StreamingQuery.scala:218]: I believe it's not yet... The general error handling framework for jvm client is not ready yet. And they only include error message and stack trace so far. I'm thinking to implement our own proto message to also include the offsets, which will be added in next PR

Comment (bogao007) [reply connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/streaming/StreamingQuery.scala:218]: I see, thanks for the clarification!

**Review by rangadi — APPROVED** [review]
LGTM. This needs to update CheckConnectJvmClientCompatibility.scala, right?

**Comment by WweiL** [comment]
> CheckConnectJvmClientCompatibility

Ah thanks! Removed it!

**Comment by WweiL** [comment]
@HyukjinKwon Can you merge this when get a change? Thank you!

**Comment by WweiL** [comment]
The `optional` in command.proto is needed or it throws:
```
[error] /home/wei.liu/oss-spark/connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/streaming/StreamingQuery.scala:210:19: value hasExceptionMessage is not a member of org.apache.spark.connect.proto.StreamingQueryCommandResult.ExceptionResult
[error]     if (exception.hasExceptionMessage) {
[error]                   ^
[error] one error found
```

**Review by amaliujia — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HeartSaVioR** [comment]
The linter build only failed which linter for scala/python got passed. The failure does not seem to be related.
https://github.com/WweiL/oss-spark/actions/runs/4805404722/jobs/8551810603

**Comment by HeartSaVioR** [comment]
Thanks! Merging to master.

**Pull request closed without merging**
